### ISSUE TITLE
issue "Preparing submission... this may take a few moments.." never run just in submitting stage 

### ISSUE THREAD

**Issue by SondosBsharat** [codalab/codabench#1708]
how to solve this issue?
![image](https://github.com/user-attachments/assets/783d2564-cd8b-48d8-be51-1e0e270bcc56)
it keeps on submitting status and never changes ,even the log file is empty 
![image](https://github.com/user-attachments/assets/a8af16d4-794e-4552-a46d-f021026d6bba)

**Comment by SondosBsharat** [comment]
I have not published it yet , I am testing it before publishing it

**Comment by ObadaS** [comment]
I see that the default queue is behaving correctly on our end, other competitions can run submissions without problems. I also ran a submission on my own test competition and everything worked correctly. 

Maybe the problem comes from the way your competition is configured ? Are you using the default docker image for competition or a custom one ?

**Comment by SondosBsharat** [comment]
![image](https://github.com/user-attachments/assets/b48675b1-681c-453e-ad71-25430092f2c4)

**Comment by SondosBsharat** [comment]
It still same issue , there are no logs at all and  the status is submitting

**Comment by ObadaS** [comment]
Can you add me as an organizer ? I'll try to do some submissions to see if mine works.
My username is `obada`

**Comment by SondosBsharat** [comment]
sure

**Comment by SondosBsharat** [comment]
I did ant it works

**Comment by ObadaS** [comment]
I ran your `predictions.zip` and it also got past the `Submitted` status. This looks like a stranger issue...
What browser are you using ?

**Comment by SondosBsharat** [comment]
chrome

**Comment by ObadaS** [comment]
I tried submitting on Chrome, and it also worked fine. Maybe you can try using Firefox, but it shouldn't change much since you can submit on my Iris test competition...

I can't add it to the leaderboard because the submission Failed (time limit exceeded), but it did finish.
As Organizer, you can check the status of all submissions on your competition by clicking on the Submission button under the competition title:
![image](https://github.com/user-attachments/assets/5cd6d4b9-392c-4327-818f-f7f90e3e453c)
You will my submissions attempts and status, as well as have some actions (like putting them on the leaderboard, rerunning it etc...)

**Comment by SondosBsharat** [comment]
I reran yours again, but it is still the same. Anyway, if others are able to run it, then no problem. I have increased the time; can you resubmit to ensure the scoring file will run? Because up until this point, it has not run yet

**Comment by ObadaS** [comment]
I submitted it again.
I see that you wrote about the `scoring.py` earlier. Is that supposed to be in the `predictions.zip` file ? The one I downloaded only contains a `.csv` file inside the `.zip`.

**Comment by SondosBsharat** [comment]
is there a specific format for that file? can you double check it ?

**Comment by ObadaS** [comment]
You can find examples here : https://github.com/codalab/competition-examples/tree/master/codabench
The wiki also has some guides to help you out : https://github.com/codalab/codabench/wiki#2-organizers

I will bring your issue with the team on our next meeting, hopefully I will have more information for you afterward.

Concerning your current file, the worker log might give you more information as to what is missing : 
```python

```

**Comment by SondosBsharat** [comment]
I tried to fix them but i am having this issue now .Is there any constraints to name the files ?  or does the submitted file  pass with its name ?
![image](https://github.com/user-attachments/assets/28bf98e1-c463-4d5f-a8cf-54e0d3ec5de0)

**Comment by SondosBsharat** [comment]
It seems that the submitted file does not trigger. 
![image](https://github.com/user-attachments/assets/8161fc57-fe26-4cbb-b91c-7d464eb6c6d8)

**Comment by ObadaS** [comment]
It seems like the problem comes from your scoring program. 

I suggest taking a look at the Iris scoring program : https://github.com/codalab/competition-examples/blob/master/codabench/iris/bundle/scoring_program/score.py

You can submit results for this competition (example : https://github.com/codalab/competition-examples/blob/master/codabench/iris/sample_result_submission.zip) , which is the same thing you are trying to do. 

Maybe the path in which your scoring program is trying the find the data is wrong ?

**Comment by SondosBsharat** [comment]
yes this is the issue. How should I deal with paths? like the submission file after it submitted how to refer to it in my [ingestion.py] file ? and after that how to pass the result to the scoring .. I tried to follow the examples but it is sill the same issue

**Comment by Didayolo** [comment]
@SondosBsharat 

Did you change path in the scoring and/or ingestion programs?
Please check the examples pointed earlier in the discussion and see if your path are different (inside `scoring.py`, `metadata`, etc.)

Another possibility is that the problem comes from zipping with directory structure.
Basically, when zipping submissions, programs, or data, you should **zip the files directly**, without directory.
Can you try this?

You can update your programs following this Wiki page: https://github.com/codalab/codabench/wiki/Resource-Management

**Comment by SondosBsharat** [comment]
These are the links in my scoring file. Are they correct? Previously, I included an ingestion file, but then I removed it since I only need the participants to submit their prediction results. However, after removing it, the submission stays in the running state until the time exceeds, or it continues running if I increase the time limit.:  f __name__ == "__main__":
    try:
        # Set up directories
        input_dir = sys.argv[1]
        output_dir = sys.argv[2]
        reference_dir = os.path.join(input_dir, "ref")
        prediction_dir = os.path.join(input_dir, "res")
        
        prediction_file = os.path.join(prediction_dir, "predictions.csv")
        ground_truth_file = os.path.join(reference_dir, "ground_truth.csv")

        # Logging setup
        logging.basicConfig(
            filename=os.path.join(output_dir, "scoring.log"),
            level=logging.INFO,
            format='%(asctime)s - %(levelname)s - %(message)s'
        )

        # Validate files
        if not os.path.exists(prediction_file):
            raise FileNotFoundError(f"Prediction file not found: {prediction_file}")
        if not os.path.exists(ground_truth_file):
            raise FileNotFoundError(f"Ground truth file not found: {ground_truth_file}")

        # Evaluate predictions
        results = evaluate_predictions(prediction_file, ground_truth_file)

        # Write results
        write_results(results, output_dir)

    except Exception as e:
        error_message = f"Error occurred: {str(e)}"
        logging.exception(error_message)
        with open(os.path.join(output_dir, "error.log"), 'w') as error_file:
            error_file.write(error_message)

**Comment by SondosBsharat** [comment]
It seems there is an issue because even if I try to include a print statement at the top of the scoring file, it does not print anything, so it is not being accessed

**Comment by Didayolo** [comment]
@SondosBsharat Any update on this? Did you manage to solve the issue?

**Comment by Didayolo** [comment]
I close the issue because it's staling. Feel free to re-open it if needed.
